src/arch/src/aarch64/regs.rs

```diff
@@ -256,7 +276,10 @@ pub fn is_system_register(regid: u64) -> bool {
 ///
 /// * `vcpu` - Structure for the VCPU that holds the VCPU's fd.
 pub fn read_mpidr(vcpu: &VcpuFd) -> Result<u64> {
-    vcpu.get_one_reg(MPIDR_EL1).map_err(Error::GetSysRegister)
+    match vcpu.get_one_reg(MPIDR_EL1) {
+        Err(err) => Err(Error::GetSysRegister(err)),
+        Ok(val) => Ok(val.try_into().expect("MPIDR register to be 64 bit"))
```

Comment:
Let's also provide a documentation to this expect, to explain that we don't expect it to happen. 
```suggestion
        // MPIDR register is 64-bit so this operation will never fail.
        Ok(val) => Ok(val.try_into().expect("MPIDR register to be 64 bit"))
```
or something similar

Reply:
done!